### PR TITLE
Reorg E2E test cases, add new E2E test cases

### PR DESCRIPTION
**Pull request by 3quanfeng**
## Description
- reorg the structure of e2e test cases, each of test cases is independent to each other, benefit is to run in parellel
- add new test cases, three-ingress-slash-sth and three-namespaces, three-ingress-slash-sth to test path based routing with backend-path-prefix, three-namespaces to test unique health probes

### CODE REVIEW

File: scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/three-namespaces/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+set -auexo pipefail
+
+echo -e "The goal of this is to ensure that containers with the same probel and same labels in 3 different namespaces have unique and working health probes"
+
+for ns in ns-x ns-y ns-z; do
+    kubectl create namespace "${ns}" || true
+
+kubectl apply -f - <<EOF
+apiVersion: apps/v1
```
Comment (akshaysngupta) [inline scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/three-namespaces/deploy.sh:11]: you can use helm instead

File: scripts/e2e/cmd/runner/helper.go [hunk 2]
```diff
@@ -93,29 +94,77 @@ func applyYaml(clientset *kubernetes.Clientset, namespaceName string, fileName s
 
 	for _, objs := range fileObjects {
 		if secret, ok := objs.(*v1.Secret); ok {
-			if _, err := clientset.CoreV1().Secrets(namespaceName).Create(secret); err != nil {
-				return err
+			nm := secret.Namespace
+			if len(nm) == 0 && len(namespaceName) != 0 {
```
Comment (akshaysngupta) [inline scripts/e2e/cmd/runner/helper.go:98]:
```golang
if len(nm) != 0 {
    ...
} else if len(namespaceName) != 0 {
    ...
} else {
    return errors.New("namespace is not defined for secrets")
}
```

Comment (3quanfeng) [reply scripts/e2e/cmd/runner/helper.go:98]: > ```go
> if len(nm) != 0 {
>     ...
> } else if len(namespaceName) != 0 {
>     ...
> } else {
>     return errors.New("namespace is not defined for secrets")
> }
> ```

In order to use namespace defined in yaml, yaml spec should have the namespace defined and the provided namespace should be empty.

File: scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/three-namespaces/deploy.sh (hunk 1 above)
Comment (3quanfeng) [reply scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/three-namespaces/deploy.sh:11]: we don't usually use the script, but It's handy when deploy it locally :)

File: scripts/e2e/cmd/runner/testdata/one-namespace-many-ingresses/three-ingresses-slash-sth/app.yaml [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+        paths:
+        - path: /kuard
+          backend:
+            serviceName: kuard
+            servicePort: 80
+---
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: ingress-d
```
Comment (akshaysngupta) [inline scripts/e2e/cmd/runner/testdata/one-namespace-many-ingresses/three-ingresses-slash-sth/app.yaml:164]: nit: four ingresses ?

File: scripts/e2e/cmd/runner/testdata/one-namespace-many-ingresses/three-ingresses-slash-sth/deploy.sh [hunk 4]
```diff
@@ -0,0 +1,197 @@
+#!/bin/bash
+
+set -auexo pipefail
+
+kubectl create namespace e2e-three-ings || true
+
+cat <<EOF | KUBECONFIG=$HOME/.kube/config kubectl apply -f -
```
Comment (akshaysngupta) [inline scripts/e2e/cmd/runner/testdata/one-namespace-many-ingresses/three-ingresses-slash-sth/deploy.sh:7]: kubectl apply -f app.yaml ?

File: scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/three-namespaces/deploy.sh (hunk 1 above)
Comment (akshaysngupta) [reply scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/three-namespaces/deploy.sh:11]: kubectl apply -f app.yaml ?

File: scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+				ns := &v1.Namespace{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: nm,
+					},
+				}
+				klog.Info("Creating namespace: ", nm)
+				_, err = clientset.CoreV1().Namespaces().Create(ns)
+				Expect(err).To(BeNil())
+			}
+			threeNamespacesYamlPath := "testdata/one-namespace-one-ingress/three-namespaces/app.yaml"
```
Comment (akshaysngupta) [inline scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:83]: one-namespace .../...three-namespace ?

Comment (3quanfeng) [reply scripts/e2e/cmd/runner/mfu_one_namespace_one_ingress_test.go:83]: one namespace one ingress is a  mapping, if you have three namespaces, and each have one ingress the mapping will still be 1 to 1

File: scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/three-namespaces/deploy.sh (hunk 1 above)
Comment (3quanfeng) [reply scripts/e2e/cmd/runner/testdata/one-namespace-one-ingress/three-namespaces/deploy.sh]: > kubectl apply -f app.yaml ?

updated

**Pull request merged**